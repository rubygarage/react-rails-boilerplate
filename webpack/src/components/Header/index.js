import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

function Header({ currentUser, currentUserProfileLink, handleSignOut }) {
  return (
    <nav className="navbar primary-color">
      <form className="form-inline justify-content-end">
        {currentUser.id ? (
          <div>
            <Link className="btn btn-primary mr-2" to={currentUserProfileLink}>
              <FormattedMessage id="header.my_profile" />
            </Link>
            <button className="btn btn-primary" onClick={handleSignOut} >
              <FormattedMessage id="header.logout" />
            </button>
          </div>
        ) : (
          <Link className="btn btn-primary" to="/sign_in">
            <FormattedMessage id="header.signin" />
          </Link>
        )}
      </form>
    </nav>
  );
}

Header.propTypes = {
  currentUser: PropTypes.shape({
    id: PropTypes.string,
    username: PropTypes.string,
    email: PropTypes.string,
  }),
  currentUserProfileLink: PropTypes.string,
  handleSignOut: PropTypes.func.isRequired,
};

export default Header;
