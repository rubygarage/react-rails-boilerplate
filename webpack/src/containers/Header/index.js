import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getCurrentUser } from 'selectors/user';
import signOut from 'actions/signout';

import HeaderComponent from 'components/Header';

export class Header extends Component {
  static propTypes = {
    currentUser: PropTypes.shape({
      id: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string,
    }),
    signOut: PropTypes.func.isRequired,
  }

  get currentUserProfileLink() {
    return `/user/${this.props.currentUser.id}`;
  }

  handleSignOut = (event) => {
    event.preventDefault();
    this.props.signOut();
  }

  render() {
    const props = {
      currentUser: this.props.currentUser,
      currentUserProfileLink: this.currentUserProfileLink,
      handleSignOut: this.handleSignOut,
    };

    return (<HeaderComponent {...props} />);
  }
}

const mapStateToProps = state => ({
  currentUser: getCurrentUser(state),
});

const mapDispatchToProps = { signOut };

export default connect(mapStateToProps, mapDispatchToProps)(Header);
