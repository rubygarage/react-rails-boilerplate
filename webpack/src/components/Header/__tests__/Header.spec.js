import React from 'react';
import Header from 'components/Header';
import { shallow } from 'enzyme';

describe('<Header />', () => {
  it('render header with sign in button', () => {
    const props = {
      currentUser: {},
      currentUserProfileLink: '',
      handleSignOut: jest.fn(),
    };

    const component = shallow(<Header {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('render header with Sign Out and Show Profile', () => {
    const props = {
      currentUser: { id: 1 },
      currentUserProfileLink: '/user/profile',
      handleSignOut: jest.fn(),
    };

    const component = shallow(<Header {...props} />);

    expect(component).toMatchSnapshot();
  });
});
