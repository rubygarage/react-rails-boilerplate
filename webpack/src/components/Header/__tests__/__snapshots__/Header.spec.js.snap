// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render header with Sign Out and Show Profile 1`] = `
<nav
  className="navbar primary-color"
>
  <form
    className="form-inline justify-content-end"
  >
    <div>
      <Link
        className="btn btn-primary mr-2"
        replace={false}
        to="/user/profile"
      >
        <FormattedMessage
          id="header.my_profile"
          values={Object {}}
        />
      </Link>
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          id="header.logout"
          values={Object {}}
        />
      </button>
    </div>
  </form>
</nav>
`;

exports[`<Header /> render header with sign in button 1`] = `
<nav
  className="navbar primary-color"
>
  <form
    className="form-inline justify-content-end"
  >
    <Link
      className="btn btn-primary"
      replace={false}
      to="/sign_in"
    >
      <FormattedMessage
        id="header.signin"
        values={Object {}}
      />
    </Link>
  </form>
</nav>
`;
